release/dist/qnap/targets.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package qnap
+
+import "tailscale.com/release/dist"
+
+func Targets(signer dist.Signer) []dist.Target {
```

Comment:
I see. The way `qbuild` (from `build-qpkg.sh`) works is by implicitly detecting `private_key` and `certificate` files in the `Tailscale` directory: https://github.com/qnap-dev/qdk2/blob/master/QDK_2.x/bin/qbuild#L974-L977
It doesn't support a remote signer IIUC, just a raw on-disk private key.

So in order to do this signing, we'd need the build server to populate the private key locally on disk, and `release/dist/qnap/pkgs.go` would need to copy that file into the `Tailscale` directory at build time. So the argument here would be `privateKeyPath, certificatePath string` instead of `dist.Signer`.

Reply:
Do we still need the `dist.Signer` piece for uploading it to the pkgs server? Or signed with the qnap signer keys is enough?

Reply:
We don't need `dist.Signer` to upload to pkgs. Theoretically, if we wanted to support `tailscale update` (and auto-updates) using our pkgs.tailscale.com packages instead of the QNAP app store, we would want `dist.Signer` to sign them.
If we only ever use their app store for updates ([like we do currently](https://github.com/tailscale/tailscale/blob/5100bdeba723728e03722d6a30b5dac4bd860b01/clientupdate/clientupdate.go#L1163)), then these `dist.Signer` signatures are not needed.

So it all depends on whether we would want to go around the QNAP app store for doing client updates.